### ISSUE TITLE
TagList shouldn't render tags as <input> elements when not in Edit mode

### ISSUE THREAD

**Issue by mirka** [Automattic/simplenote-electron#844]
The TagList component in the NavigationBar is rendering tags as `<input>` elements, even when they're not in editing mode. When not in editing mode, the tags act like links, not input fields.

There are two problems with this:

1. It is unsemantic HTML and not good for accessibility.
1. An Undo/Redo context menu will be shown on right click, which doesn't make sense for links.

They should be `<a>` (or `<button>`?) elements by default, and only become `<input>` when in editing mode.
